woocommerce/woocommerce-gutenberg-products-block
# Fix: Mini Cart: Ability to choose the drawer behavior when user add products to cart

Fixes #6026 

This PR:

- Adds a new attribute named ~`openDrawerOnAddedToCart` as a toggle~. I changed this option to `addedToCartBehavior` using Select Control in 2a842b1c40e86ec70e143a827c5f7a52c629f766.
- Only opens the drawer when the added to cart behavior is set to `Open the drawer`.
- Changes default behavior to **not** open the drawer when a product is added to the cart.

To the reviewers: '
- I'm wondering which default behavior makes more sense here. The current behavior is opening the drawer every time a product is added to the cart and it interrupts the shopping flow.
- The copy for options of the new attribute may be not so good, any feedback on that is more than welcome.

### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

1. Add Mini Cart to the Header template part.
2. Open the Block Setting sidebar, see the new setting pane: `Mini Cart Settings`.
3. See it set to `Do not show anything` as default.
4. Save changes.
5. On a new tab, go to the site front end.
6. Add a product to the cart from the `All products` block. Don't see the drawer open.
7. Back to the Site Editor, change the behavior to `Open drawer`. Save changes.
8. On a new tab, go to the site front end.
9. Add a product to cart, see the drawer open.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
